### ISSUE TITLE
Accept an empty $ADB_DEVICE_ARG when $ANDROID_SERIAL is set

### ISSUE THREAD

**Issue by jonasfa** [calabash/calabash-android#778]
`calabash-android` fails when there are multiple devices and the `$ADB_DEVICE_ARG` environment variable is not set.

The Android tools already have an environment variable to set the target devicefor `adb`, which is `$ANDROID_SERIAL`.

We probably shouldn't remove `$ADB_DEVICE_ARG` completely for compatibility with existing usages of `calabash-android`, but it's desirable to follow the standard and use `$ANDROID_SERIAL` when it is set. We don't need to user `-s [the-serial]` in this case.

**Comment by TobiasRoikjer** [comment]
I don't think this is necessarily a good idea. Calabash is a framework on its own, and we don't always adapt the naming conventions/arguments that the native framework does.

Consider this:
After setting the ENV variable, somebody wants to change the default device serial. The ENV can therefore no longer be given directly to the spawned subprocess (ADB).
This can be a likely scenario for Calabash in a CI system.

In general, we don't want to ever change the value of an environment variable (some older code in the project does not always conform to this).

Secondly, implemented in Calabash 2.0, we want to prefix out ENV variables with CAL.

**Comment by jonasfa** [comment]
It's not a convention. calabash-android is effectively duplicating a feature from the official Android tools.

calabash-android wouldn't set the env var. That's the user's responsibility, as it already is with `$ADB_DEVICE_ARG`

**Comment by TobiasRoikjer** [comment]
But what if they differ?

I understand your reasoning, but I don't see a reason why we should extend Calabash to support this feature.

**Comment by jonasfa** [comment]
If 1. being compatible with your target platform's official tools and 2. not duplicating a feature from a tool you already depend on are not good reasons, I'm afraid I don't have any more arguments.

**Comment by TobiasRoikjer** [comment]
I appreciate your comments and suggestions very much, but I don't think this change is a good idea.

So there would be two arguments for adding this:
1) Either we support both ENV variables
2) We use the ANDROID_SERIAL variable, and deprecate ADB_DEVICE_ARG

for 1) I don't think it is a good idea to have two environment variables that do the same thing, as one would have to take precedence, or we would have to fail if both were set. I think it would also be the cause of confusion.

for 2) I don't think we should use the same ENV variable as the Android dev tools. We want to prefix ENV variables with CAL and name them so it is easy for users to understand. For Calabash 2.0, we have named it CAL_DEVICE_ID.

The current name of the ENV variable is not good either, but I don't think it warrants being renamed for 1.0.

In general, Calabash does not adopt any naming convention/standard of the target native framework without a reason to do so. We want to have all the power that the native frameworks provide, but often expose them as our own API.

That also means we do not want users to care or have to be aware of ADB's existence unless they want to. Many users are not familiar with the mobile platform they are targeting.

For iOS, for example, we are not using instruments anymore since iOS removed it and Apple recommend the tooling driving XCUITests. Many users are likely unaware of this, and Calabash abstracts the tooling for them. It is, of course, extremely unlikely that Calabash would no longer use ADB.

Of course these are goals that we apply on a case by case basis, and much of our old code/naming does not follow this.
